### ISSUE TITLE
Geometry_Engine: CleanPolyline variable name

### ISSUE THREAD

**Issue by michaldengusiak** [BHoM/BHoM_Engine#1173]
#### Description:

 I recommend we replace name from distanceTolerance to minimumSegmentLength as below ,I will keep right now as we need to fixed also in  `XML Toolkit`. We discussed with @pawelbaran and @KonradStolarski  this is not tolerance but minimumSegmentLength so it is confusing for users. In case we have some point along line in the close distance we need to remove them for Environment it is 0.02. 

` static Polyline CleanPolyline(this Polyline polyline, double angleTolerance = Tolerance.Angle, double minimumSegmentLength = 0.02)`

#### How to replicate:

clarification of documentation and naming

**Comment by pawelbaran** [comment]
Agreed to 50%, I would go for changing the name from `distanceTolerance` to `minimumSegmentLength`, but I would leave the latter without a default value as 0.02 sounds quite problem-specific.

**Comment by michaldengusiak** [comment]
I am happy with this solution but I would prefer to have my default, 
one item @FraserGreenroyd  we just need to check where we use this method to make sure that input is exposed to the user for all nodes that we use this as we using this in quite a few places

**Comment by rwemay** [comment]
Agree that 0.02 as a default minimum segment length is too problem specific. Shouldn't this be 0 or the global default (x10-6?). As this is a global/central engine method, it might also be good to be more specific with the name - what does 'clean' mean - remove short segments? (apologies have not looked at the actual method if there is one!!).

**Comment by pawelbaran** [comment]
I do not like the method name either tbh, just did not want to be the bad guy ;-) I would still go for no default as 1e-6 is way too low for any engineering purposes I can imagine (apart from removing duplicate nodes), ergo the users may be confused. Or am I confusing it? 🐒

**Comment by FraserGreenroyd** [comment]
I think perhaps confusing it at the latter part @pawelbaran but easily done. The method name makes sense for what the method is doing (to me at least, but then I wrote the method so of course I'm happy with it's name!).

I agree, default segment length will either come from `BH.oM.Geometry.Tolerance` as standard with option to override, or will have no default. Default of `0.02` does not make any immediate engineering sense and risks confusing users because it will be different to existing conventions within the code base.

@rwemay `CleanPolyline` does have an implementation within the code, it tidies up polylines to remove potentially unnecessary vertices (e.g. three points in a straight line, the middle point may be unnecessary and cause confusion in later simulation toolkits, so the middle point is removed). Currently it is used an XML and TAS toolkits which is why it needs to sit within the central geometry engine to avoid code duplication/maintenance.

Long story short:
 - Rename variable to remove 'tolerance'
 - Default will not be 0.02. Either something from `BH.oM.Geometry.Tolerance` or `null`.

**Comment by pawelbaran** [comment]
I am OK with the above!

**Comment by michaldengusiak** [comment]
yes, this sounds fine, 
*we just need to exposed this tolerance to top-level.. so nodes in XMLSerailization and TAS
for default segment length `BH.oM.Geometry.Tolerance` make no sense so better use `null`

**Comment by rwemay** [comment]
Thanks @FraserGreenroyd all makes sense, I would make the above clear in the description if not already. This is a good one for multidis too as useful for structural eng applications for example. It does sound like its doing two things ‘RemoveLeastSignificantVertices’ and ‘RemoveShortSegments’(?) both of which could be exposed as well as a CleanPolyline (or SimplifyPolyline perhaps more appropriate) which calls the former.

**Comment by FraserGreenroyd** [comment]
@rwemay agreed, I think that would be a far more powerful and beneficial workflow, offering options on which 'clean' you do, but having power to clean with both for when you need it. I'm happy with the method names you've used, so I'll get a PR up later today for us all to take a look at.
